main.c

```diff
@@ -16,28 +16,53 @@
 
 #include "tinywav.h"
 
-#define NUM_CHANNELS 1
```

Comment:
My overall comment on this file is that I see where you're going with it, but IMO you took a simple example and made it complicated. Though its good that there is an example of reading now.

For me this example is supposed to be a _simple_ example that also acts as a kind of test. People can surely figure out their own error checking.

Reply:
I see what you mean --- I figured the example code was meant to go in the readme and saw this as a 'full loop' test file.

I agree with you, it's more complicated than it should be...  the reason I is (I confess) that I copy-pasted parts of the code from a project where I'm using tinywav.

I will simplify this file as much as I can 👍 
The writer will always write a whole number of blocks, so the resulting file might be a fraction of a block longer than the input (additional samples are zero). This will greatly simplify the code.